src/test/javascript/portal/cart/WpsDownloadHandlerSpec.js

```diff
@@ -77,25 +74,22 @@ describe('Portal.cart.WpsDownloadHandler', function () {
                     }
                 ])
             };
+
             testHandlerParams = {
                 emailAddress: 'bob@example.com'
             };
-
-            url = clickHandler(testCollection, testHandlerParams);
-
-            json = jsonFromUrl(url, expectedUrlStart);
-
         });
 
         it('builds the correct URL', function() {
+            var url = clickHandler(testCollection, testHandlerParams);
 
             expect(url).toStartWith(expectedUrlStart);
 
-            expect(json.typeName).toBe('layer_name');
-            expect(json.emailAddress).toBe('bob@example.com');
-            expect(json.url).toBe('geoserver_url');
-            expect(json.cqlFilter).toBe('Geometry Cql AND Salinity Cql');
+            var decodedUrl = Ext.urlDecode(url);
+            expect(decodedUrl['server']).toBe('geoserver_url');
+            expect(decodedUrl['jobParameters.typeName']).toBe('layer_name');
+            expect(decodedUrl['jobParameters.email.to']).toBe('bob@example.com');
+            expect(decodedUrl['jobParameters.cqlFilter']).toBe('Geometry Cql AND Salinity Cql');
```

Comment:
This could also be written using the `toHaveParameterWithValue()` matcher like so:
``` javascript
expect(url).toStartWith(expectedUrlStart);
expect(url).toHaveParameterWithValue('server', 'geoserver_url');
expect(url).toHaveParameterWithValue('jobParameters.typeName', 'layer_name');
expect(url).toHaveParameterWithValue('jobParameters.email.to', 'bob@example.com');
expect(url).toHaveParameterWithValue('jobParameters.cqlFilter', 'Geometry Cql AND Salinity Cql');
```